ui-admin/src/App.tsx

```diff
@@ -50,42 +52,49 @@ mixpanel.register({ application: 'ADMIN_UI' })
 /** container for the app including the router  */
 function App() {
   return (
-    <ConfigProvider>
-      <ConfigConsumer>
-        { config =>
-          <AuthProvider {...getOidcConfig(config.b2cTenantName, config.b2cClientId, config.b2cPolicyName)}>
-            <UserProvider>
-              <div className="App d-flex flex-column min-vh-100">
-                <IdleStatusMonitor maxIdleSessionDuration={30 * 60 * 1000} idleWarningDuration={5 * 60 * 1000}/>
-                <ReactNotifications />
-                <BrowserRouter>
-                  <ScrollToTop/>
-                  <Routes>
-                    <Route path="/">
-                      <Route path="/system/status" element={<StatusPage/>}/>
-                      <Route element={<ProtectedRoute>
-                        <NavContextProvider><PageFrame config={config}/></NavContextProvider>
-                      </ProtectedRoute>}>
-                        <Route path="populate/*" element={<PopulateRouteSelect/>}/>
-                        <Route path="logEvents/*" element={<LogEventViewer/>}/>
-                        <Route path="users/*" element={<AdminUserRouter/>}/>
-                        <Route path="integrations/*" element={<IntegrationDashboard/>}/>
-                        <Route path=":portalShortcode/*" element={<PortalProvider><PortalRouter/></PortalProvider>}/>
-                        <Route index element={<HomePage/>}/>
-                      </Route>
-                      <Route path="privacy" element={<PrivacyPolicyPage />} />
-                      <Route path="terms" element={<InvestigatorTermsOfUsePage />} />
-                      <Route path="*" element={<div>Unknown page</div>}/>
-                    </Route>
-                    <Route path='redirect-from-oauth' element={<RedirectFromOAuth/>}/>
-                  </Routes>
-                </BrowserRouter>
-              </div>
-            </UserProvider>
-          </AuthProvider>
-        }
-      </ConfigConsumer>
-    </ConfigProvider>
+    <ApiProvider api={previewApi('', '')}>
```

Comment:
we may want to get slightly more sophisticated here and periodically poll the maintenance mode endpoint in case users are actively using the system when we enter maintenance mode. as it stands right now, only a hard-refresh would reload the latest settings